src/main/java/com/hermesworld/ais/galapagos/topics/TopicMetadata.java

```diff
@@ -66,6 +71,16 @@ public TopicMetadata(TopicMetadata original) {
         this.criticality = original.criticality;
         this.messagesPerDay = original.messagesPerDay;
         this.messagesSize = original.messagesSize;
+        this.producers = original.producers;
+    }
+
+    public void addProducers(List<String> producers) {
```

Comment:
add and remove logic shall never be in a Data class like this. Handle this in TopicServiceImpl and just SET the contained list in this class accordingly (really SET the list each time, as it could be - and should be - immutable).